onnxruntime/core/optimizer/constant_folding.cc

```diff
@@ -20,66 +20,63 @@ Status ConstantFolding::ApplyImpl(Graph& graph, bool& modified, int graph_level)
     if (!node) {
       continue;
     }
-
-    ORT_RETURN_IF_ERROR(Recurse(*node, modified, graph_level));
-
-    InitializedTensorSet constant_inputs;
-
-    // Check if constant folding can be applied on this node.
-    if (!graph_utils::IsSupportedProvider(*node, GetCompatibleExecutionProviders()) ||
-        excluded_op_types_.find(node->OpType()) != excluded_op_types_.end() ||
-        // constant folding does not support executing a node that includes subgraphs (control flow operators,
-        // such as If/Loop/Scan, fall into this category). individual nodes in the subgraph will be processed
-        // by the Recurse call above
-        node->ContainsSubgraph() ||
-        // if the node output is in the graph output, we will get a graph with no nodes.
-        // TODO check if this is allowed in ONNX and ORT.
-        graph.IsNodeOutputsInGraphOutputs(*node) ||
-        !graph_utils::AllNodeInputsAreConstant(graph, *node, constant_inputs)) {
+    try {
```

Comment:
Should all optimizers have try/catch to ensure we don't fail because an optimizer failed?

Reply:
Yes. I have a bigger and better plan. We can do it later, after the 1.0 release.